apps/web/pages/api/book/event.ts

```diff
@@ -10,12 +10,23 @@ import getIP from "@calcom/lib/getIP";
 import { piiHasher } from "@calcom/lib/server/PiiHasher";
 import { checkCfTurnstileToken } from "@calcom/lib/server/checkCfTurnstileToken";
 import { defaultResponder } from "@calcom/lib/server/defaultResponder";
-import prisma from "@calcom/prisma";
+import type { TraceContext } from "@calcom/lib/tracing";
+import { distributedTracing } from "@calcom/lib/tracing/factory";
+import { prisma } from "@calcom/prisma";
 import { CreationSource } from "@calcom/prisma/enums";
 
-async function handler(req: NextApiRequest & { userId?: number }) {
+async function handler(req: NextApiRequest & { userId?: number; traceContext: TraceContext }) {
   const userIp = getIP(req);
 
+  const traceContext = distributedTracing.updateTrace(req.traceContext, {
+    eventTypeId: req.body?.eventTypeId?.toString() || "null",
```

Comment:
I would suggest updating the types for meta to include null/number/boolean even undefined maybe for the Record values so that this could be cleaned up to just
```
{
  eventTypeId: req.body?.eventTypeId
}
```